### PR TITLE
Keep template vals flag for new command [WIP (Work In Progress)]

### PR DESCRIPTION
**Pull request by therishidesai**
solves #49

### CODE REVIEW

File: cmd/new.go [hunk 1]
```diff
@@ -59,4 +59,5 @@ $ cfg new myTempl myInstance.json -- --random`,
 
 func init() {
 	rootCmd.AddCommand(newCmd)
+	newCmd.Flags().BoolVarP(&newCfg.KeepTempVals, "keep-temp-vals", "", false, "keeps values from the template")
```
Comment (thomasdfischer) [inline cmd/new.go]: switch to `--use-templ-values` or `--same-as-templ`

File: new/config.go [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package new
 
 type Config struct {
-	TemplName string
-	Targets   []string
+	TemplName    string
+	Targets      []string
+	KeepTempVals bool
```
Comment (thomasdfischer) [inline new/config.go]: Rename to `DoUseTemplValues` or `DoSameAsTempl` (main point being `Do` prefix)

File: new/new.go [hunk 3]
```diff
@@ -13,13 +13,13 @@ import (
 )
 
 func MustNew(cfg *Config) {
-	if err := New(cfg.TemplName, cfg.Targets); err != nil {
+	if err := New(cfg.TemplName, cfg.Targets, cfg.KeepTempVals); err != nil {
```
Comment (thomasdfischer) [inline new/new.go]: DoWhatever

File: new/new.go [hunk 4]
```diff
@@ -13,13 +13,13 @@ import (
...
 		fmt.Fprintf(os.Stderr, "error: %v\n", err)
 		os.Exit(1)
 	}
 }
 
-func New(templName string, targets []string) error {
+func New(templName string, targets []string, keepVals bool) error {
```
Comment (thomasdfischer) [inline new/new.go]: refactor to just pass the config object to this function, instead of having so many params

File: new/new.go [hunk 5]
```diff
@@ -44,13 +44,22 @@ func New(templName string, targets []string) error {
...
-	args := append([]string{"init", "-s", templSchemaPath}, targets...)
-	cmd := exec.Command("schema", args...)
-	if err := cmdrunner.PipeTo(cmd, os.Stdout, os.Stderr); err != nil {
-		return err
+	if !keepVals {
+		args := append([]string{"init", "-s", templSchemaPath}, targets...)
+		cmd := exec.Command("schema", args...)
+		if err := cmdrunner.PipeTo(cmd, os.Stdout, os.Stderr); err != nil {
+			return err
+		}
```
Comment (thomasdfischer) [inline new/new.go:56]: need to `return` at the end of this `if` statement, or the rest of the code will erroneously execute

File: new/new.go [hunk 6]
```diff
@@ -44,13 +44,22 @@ func New(templName string, targets []string) error {
...
 	}
 
+	templFile, err := os.Open(templObj.FilePath)
```
Comment (thomasdfischer) [inline new/new.go]: check the error

File: new/new.go [hunk 7]
```diff
@@ -44,13 +44,22 @@ func New(templName string, targets []string) error {
...
+
 	for _, target := range targets {
+		if keepVals {
+			f, err := os.Create(target)
```
Comment (thomasdfischer) [inline new/new.go]: check the errors

File: new/config.go (hunk 2 above)
Comment (thomasdfischer) [reply new/config.go]: (and mirror the name of the flag)

File: new/new.go (hunk 5 above)
Comment (thomasdfischer) [reply new/new.go:56]: disregard

File: cmd/new.go [hunk 8]
```diff
@@ -59,4 +59,5 @@ $ cfg new myTempl myInstance.json -- --random`,
 
 func init() {
 	rootCmd.AddCommand(newCmd)
+	newCmd.Flags().BoolVarP(&newCfg.DoUseTemplVals, "use-temp-values", "", false, "keeps values from the template")
```
Comment (thomasdfischer) [inline cmd/new.go]: Forgot to mention yesterday --- this should be `true` by default

**Comment by therishidesai** [comment]
So now this is how it works:
```
/Confbase/test  ᐅ ./../cfg/cfg new config config_prod_na5.toml --default                 
/Confbase/test (master ✘)✹ ᐅ cat config_prod_na5.toml                    
{
    "color": "",
    "height": 0
}
/Confbase/test (master ✘)✹ ᐅ ./../cfg/cfg new config config_prod_na5.toml
/Confbase/test (master ✔) ᐅ cat config_prod_na5.toml                                      
color = "blue"
height = 1.0%                                                        
```

**Pull request merged**